apache/cloudstack
# Generate cloud-init multipart user data for template append policy

### Description

This PR introduces the concept of a user data provider to handle the user data concatenation in case a template has an APPEND policy and a VM also has user data.

The cloud-init user data provider is added as the default provider. It concatenates the template and VM user data as a multipart MIME message

Supported data formats: https://cloudinit.readthedocs.io/en/latest/explanation/format.html#user-data-formats

### Types of changes


### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale


#### Bug Severity



### Screenshots (if appropriate):


### How Has This Been Tested?
- Add user data for different types (cloud-config, bash scripts)
- Register a template and link userdata with APPEND policy
- Deploy VM from the template and link another existing userdata
- Verify the generated user data is a multipart MIME message
